scripts/us_bea/states_gdp/import_industry_data_and_gen_mcf.py

```diff
@@ -20,148 +20,143 @@
+        df = df[df['GeoName'].isin(self._US_STATES)]
 
         # Gets columns that represent quarters, e.g. 2015:Q2, by matching
         # against a regular expression.
-        all_quarters = [q for q in df.columns if re.match(r"....:Q.", q)]
+        all_quarters = [q for q in df.columns if re.match(r"\d\d\d\d:Q\d", q)]
 
         # Convert table from wide to long format.
         df = pd.melt(df, id_vars=['GeoFIPS', 'IndustryClassification'],
                      value_vars=all_quarters,
                      var_name='Quarter')
 
-        qtr_month_map = {
-            'Q1':'03',
-            'Q2':'06',
-            'Q3':'09',
-            'Q4':'12'
-        }
-
-        def date_to_obs_date(date):
-            """Converts date format e.g. 2005:Q3 to e.g. 2005-09."""
-            return date[:4] + "-" + qtr_month_map[date[5:]]
-        df['Quarter'] = df['Quarter'].apply(date_to_obs_date)
-
-        def convert_geoid(fips_code):
-            """Creates GeoId column. We get lucky that Data Commons's geoIds
-            equal US FIPS state codes.
-            """
-            fips_code = fips_code.replace('"', "")
-            fips_code = fips_code.replace(" ", "")
-            return "geoId/" + fips_code[:2]
-        df['GeoId'] = df['GeoFIPS'].apply(convert_geoid)
+        df['Quarter'] = df['Quarter'].apply(self._date_to_obs_date)
+        df['GeoId'] = df['GeoFIPS'].apply(self._convert_geoid)
 
         df = df[df['IndustryClassification'] != "..."]
 
-        def convert_industry_class(naics_code):
-            """Filters out aggregate NAICS codes and assigns them their Data
-            Commons codes.
-            """
-            if naics_code == "321,327-339":
-                return "JOLTS_320000"
-            if naics_code == "311-316,322-326":
-                return "JOLTS_340000"
-            return naics_code.replace("-", "_")
-        df['NAICS'] = df['IndustryClassification'].apply(convert_industry_class)
+        df['NAICS'] = df['IndustryClassification'].apply(self._convert_industry_class)
+        df['value'] = df['value'].apply(self._value_converter)
+        df = df[df['value'] >= 0]
 
-        self.df = df.drop(["GeoFIPS","IndustryClassification"], axis=1)
+        self.clean_df = df.drop(["GeoFIPS", "IndustryClassification"], axis=1)
 
-    def download_data(self):
-        """Downloads zip, extracts the desired CSV, and puts it into a DF."""
-        # Open zip file from link.
-        resp = urlopen(self.ZIP_LINK)
-
-        # Read the file, interpret it as bytes, and create a ZipFile instance
-        # from it for easy handling.
-        zip_file = zipfile.ZipFile(io.BytesIO(resp.read()))
+    @staticmethod
+    def _value_converter(val):
+        """Converts value to float type, and filters out missing values. Missing
+        values are marked by a latter enclosed in parentheses, e.g., "(D)".
+        """
+        if isinstance(val, float):
+            return val
+        if '(' in val or ')' in val:
```

Comment:
Do we need to check that val is an str?

Reply:
Done. Checking for is not str above instead.